ingest/census_acs.py

```
@@ -368,12 +371,14 @@ def _get_cached_tract_data(self, state_code: str, county_code: str,
         # Check if we have recent cached data
-        cursor.execute('''
+        cursor.execute(f'''
             SELECT variable_data FROM census_tract_cache 
             WHERE state_code = ? AND county_code = ? AND tract_code = ?
-                AND datetime(fetched_at) > datetime('now', '-{} hours')
+              AND data_vintage = 2022
+              AND datetime(fetched_at) > datetime('now', '-{max_age_hours} hours')
+              AND (expires_at IS NULL OR expires_at > CURRENT_TIMESTAMP)
             ORDER BY fetched_at DESC LIMIT 1
-        '''.format(max_age_hours), (state_code, county_code, tract_code))
+        ''', (state_code, county_code, tract_code))
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)